dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLlvm.java

```diff
@@ -1487,5 +1513,15 @@ public <T extends MdNode> Optional<T> getField(String fieldName) {
             return Optional.empty();
         }
     }
+    
+    // ----------------------------------------------------------------------------------------------------------------
+    // Helper to parse inline asm code
+    private Optional<List<Event>> tryParse(ParserAsm parser, CharStream asmCode) throws ProgramProcessingException{
```

Comment:
Why the `ProgramProcessingException`? I think in the parser we should only use `ParsingException`.

Reply:
I think you can add a new exception type or use `UnsupportedOperationException` for unimplemented cases. That's the canonical choice at least

Reply:
The current implementation is wrong. Unless the code uses armv8 asm (which is the first parser in your list), you will throw and exception and simply treat the whole call as a non-det value. This is the reason why all asm tests except armv8 are failing